platform/linuxbsd/detect.py

```diff
@@ -90,7 +90,7 @@ def configure(env):
     if env["target"] == "release":
         if env["optimize"] == "speed":  # optimize for speed (default)
             env.Prepend(CCFLAGS=["-O3"])
-        else:  # optimize for size
+        elif env["optimize"] != "none":  # optimize for size
```

Comment:
```suggestion
        elif env["optimize"] == "size":  # optimize for size
```
No optimization flags are going to be applied with `none` as it won't be handled.